### ISSUE TITLE
Amiga 600 Vampire 2.2 chip memory test error.

### ISSUE THREAD

**Issue by Tallguy297** [keirf/amiga-stuff#59]
When I ran your Amiga Test Kit program on my Amiga 600 with Vampire 2.2 and tested chip memory I got this...

![Memory_Error](https://user-images.githubusercontent.com/49935683/158490303-d147c415-2e14-45e4-b371-3f2896707e1d.jpg)

I pulled the Vampire card out and ran your test program again.
No errors in chip memory on all tests.

**Comment by keirf** [comment]
If Vampires generally screwed up the chip memory test I would hear about it all the damn time.

**Comment by keirf** [comment]
So, I feel something is awry with your Vamp. When you reinstall(ed) is the error identical? Is the error *always* identical appearance to this?

**Comment by Tallguy297** [comment]
I thought about this at first. I took the Vampire out of the Amiga, put in the original kickstart 3.2 rom.
The main reason is that I had flashed the Vampire 2 with the Amiga kickstart 3.2.1 rom and it seems redundant having both kickstart and flashed rom in a Amiga 600.

Booted from the ADF image Amiga Test Kit and tested the memory region from 00000000 to 001FFFFF.
I received no error reports. I pulled of the RTC clock module, just in case that was giving me problems and I still didn't get any errors.
It is only when I put the Vampire 2.2 card on then I received errors when scanning chip memory.
I double checked that the Vampire card was seated correctly.
I also checked the Vampire memory region and did not receive any errors.

I also used another program MBR-Test2 and made it run for 4 passes on both chip & expansion memory and received no errors.

My chip memory expansion is an A601 2MB expansion with RTC Clock and an extra clock port for USB expansion.

I am assuming I am getting this error because of memory speed issues. The Amiga's chip memory was never designed to be overclocked. (I am probably wrong in this statement).

It is no big issue. For the most part my Amiga 600 & Vampire 2.2 work very well. I have not experienced any Gurus and all the RTG games, like Duke Nukem, Diablo & Quake work very well.

Thank you...

**Comment by keirf** [comment]
Maybe someone else with a Vampire 2.2 can test this. I'll ask around.

**Comment by Tallguy297** [comment]
I forgot to mention that the error only appears when scanning memory region `100000 - 1FFFFF` expansion memory.

Thank you.

**Comment by Tallguy297** [comment]
I purchased a brand new Amiga 600 A601 memory expansion from AmigaKit and it arrived today (Monday 10th Oct 2022).
I performed a memory test on the chip memory region 00000000 to 001FFFFF and received the same errors as initially described.
When I pulled the Vampire 2.2 card out and tested the memory, there were no errors.

Strange??

**Comment by keirf** [comment]
I'm confused. Did you not have a chip expansion to 2MB previously? The Vampire does not expand the A600's default 1MB chip.

Or are you saying the issue is independent of the RAM expansion itself, because you've tested two? This is unsurprising since the RAM expansions test okay when the Vampire is removed. So the issue resides with the Vampire.

**Comment by Tallguy297** [comment]
Sorry, I should have elaborated.
Quite some time ago I purchased the A601 memory expansion together with a Rapid Road USB expansion interface.
When I installed the USB expansion I made the common mistake of not putting the 20 pin cable on the A601 clock port the right way around and I ended up damaging the USB expansion interface.

Since then I purchased the Vampire 2.2 accelerator to replace my Furia accelerator.

I some problems with my keyboard connection so I ran your AmigaTestKit to test all the hardware. It was then I discovered the memory fault as initially described.
I thought this might have been caused by the USB expansion that I had previously installed and caused hardware damage to the A601 memory expansion.
So to be on the safe side I removed the Vampire accelerator and tested the 2Mb chip memory region using your software. I didn't receive any errors.

After a long hard think I decided to replace the memory expansion, so I purchased a new one from AmigaKit.
I performed the same 2MB chip memory test without the Vampire card and did not receive any errors.
It is only after I installed the Vampire 2 card that I received identical errors as before and in the same memory region as pictured.
I have performed numerous memory tests on the Vampire accelerator expansion memory and your software cannot detect any faults.

Thank you...

**Comment by Tallguy297** [comment]
Hello again...

I purchased the new MantiCore V4 for the Amiga 600, which is a great improvement over the Vampire 2.2 card.
This new card outputs AGA, ECS, & RTG modes with sound over the HDMI output. So need for an upscaler.

I ran your AmigaTestKit and I got the same results as with the Vampire 2.2 card.
The same memory fault in the 2MB expansion memory in exactly the same memory region.
Weird??

**Comment by keirf** [comment]
Perhaps the problem is with your 600's chip memory expansion? It is the common element here.

**Comment by keirf** [comment]
Ah you did try another chip ram expansion. Weird indeed then. However I very much doubt ATK is at fault in reporting an error in the middle of chip ram.

**Comment by keirf** [comment]
Possibly impractical, but try transplanting to another 600 if you can.

**Comment by Tallguy297** [comment]
I thought about that. I did buy a new A601 memory expansion for AmiKit and got the results as noted above.
FYI, this does not happen if I remove the Vampire card and install the Kickstart 3.2 rom.
Could be a problem with the Vampire Card. I have raised this issue with Apollo Core and they are just as puzzled as I am.
It's being investigated. However it does not affect the overall operation. I have execute other memory testing problems like MBR-2 Memtest and everything appears normal.

I was thinking that it could be a memory clock issue?

**Comment by keirf** [comment]
I also thought on reflection escalating to Apollo Core makes most sense. They should have the hardware to do some broader testing and see if this is a wider issue. And if it is, they're the ones on the hook. If it's not, at least you know that swapping out even more hardware should fix it.
